source/MaterialXGenGlsl/Nodes/PositionNodeGlsl.cpp

```diff
@@ -8,83 +8,90 @@ ShaderNodeImplPtr PositionNodeGlsl::create()
     return std::make_shared<PositionNodeGlsl>();
 }
 
-void PositionNodeGlsl::createVariables(const ShaderNode& node, ShaderGenerator& /*shadergen*/, Shader& shader_)
+void PositionNodeGlsl::createVariables(const ShaderNode& node, GenContext&, Shader& shader) const
 {
-    HwShader& shader = static_cast<HwShader&>(shader_);
+    ShaderStage vs = shader.getStage(HW::VERTEX_STAGE);
+    ShaderStage ps = shader.getStage(HW::PIXEL_STAGE);
 
-    shader.createAppData(Type::VECTOR3, "i_position");
+    addStageInput(HW::VERTEX_INPUTS, Type::VECTOR3, "i_position", vs);
 
     const ShaderInput* spaceInput = node.getInput(SPACE);
-    string space = spaceInput ? spaceInput->value->getValueString() : EMPTY_STRING;
-    if (space == WORLD)
+    const int space = spaceInput ? spaceInput->getValue()->asA<int>() : -1;
```

Comment:
Hmm, this is also a "spaces handling change" that I don't recognize.. And I agree the default value should be OBJECT_SPACE instead of -1. The value -1 will give the same result below but using OBJECT_SPACE instead would be more clear.